sycl/source/detail/devicelib/CMakeLists.txt

```diff
@@ -0,0 +1,68 @@
+# Place device libraries near the libsycl.so library in a build
+# directory
+if (MSVC)
+    set(binary_dir "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
+else()
+    set(binary_dir "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+endif()
+
+set(clang $<TARGET_FILE:clang>)
+
+set(compile_opts
+    # suppress an error about SYCL_EXTERNAL
+    -Wno-error=sycl-strict -Wno-sycl-strict
+    # for CL/__spirv/spirv_vars.hpp
+    -I${sycl_inc_dir})
+
+if (MSVC)
+    set(devicelib-obj-file ${binary_dir}/libsycl-msvc.o)
+    add_custom_command(OUTPUT ${devicelib-obj-file}
+                       COMMAND ${clang} -fsycl -c
+                               ${compile_opts}
+                               ${CMAKE_CURRENT_SOURCE_DIR}/msvc_wrapper.cpp
+                               -o ${devicelib-obj-file}
+                       MAIN_DEPENDENCY msvc_wrapper.cpp
+                       DEPENDS wrapper.h clang
+                       VERBATIM)
+else()
+    set(devicelib-obj-file ${binary_dir}/libsycl-glibc.o)
+    add_custom_command(OUTPUT ${devicelib-obj-file}
+                       COMMAND ${clang} -fsycl -c
+                               ${compile_opts}
+                               ${CMAKE_CURRENT_SOURCE_DIR}/glibc_wrapper.cpp
+                               -o ${devicelib-obj-file}
+                       MAIN_DEPENDENCY glibc_wrapper.cpp
+                       DEPENDS wrapper.h clang
+                       VERBATIM)
+endif()
```

Comment:
```
if (MSVC)
    set(devicelib-obj-id "msvc")
else()
    set(devicelib-obj-id "glibc")
endif()

set(devicelib-obj-file ${binary_dir}/libsycl-${devicelib-obj-id}.o)
add_custom_command(OUTPUT ${devicelib-obj-file}
                  COMMAND ${clang} -fsycl -c
                          ${compile_opts}
                          ${CMAKE_CURRENT_SOURCE_DIR}/${devicelib-obj-id}_wrapper.cpp
                           -o ${devicelib-obj-file}
                   MAIN_DEPENDENCY ${devicelib-obj-id}_wrapper.cpp
                   DEPENDS wrapper.h clang
                   VERBATIM)
```

Reply:
Frankly, I'd prefer to keep the current code as is. When we add more libraries (e.g. libstdc++), we may want to add them only to the `not msvc` block.

Alternatively, we can define a CMake function that takes output, source file, arguments,  list of dependencies, and then forms an add_custom_command. My cmake-kung-fu skill is not good enough to implement this.

Reply:
Ok, this is not critical. I agree that we can reduce code duplication once we have a few more libraries